### PR TITLE
Homework2 completed

### PR DESCRIPTION
**Pull request by Innokentiy945**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        <profile>
+            <id>FunctionalTest2</id>
+            <activation>
+                <activeByDefault>false</activeByDefault>
+            </activation>
+            <properties>
+                <suite.xml>resources/FunctionalTest2.xml</suite.xml>
+            </properties>
+        </profile>
+    </profiles>
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply pom.xml]: Fixed. Tests run correctly.

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+public class FunctionalTest1 {
+    private WebDriver driver;
+    private String url = "https://jdi-testing.github.io/jdi-light/index.html";
+
+    @BeforeSuite
+    public void setUp() {
+        WebDriverManager.chromedriver().setup();
+        driver = new ChromeDriver();
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Fixed. For now this in the @BeforeMethod

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    @BeforeTest
+    public void profileSetup() {
+        driver.manage().window().maximize();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Fixed. For now this is in @BeforeMethod

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+
+        //2. Assert Browser title
+        String currentTitle = driver.getTitle();
+        softAssert.assertEquals(currentTitle, "Home Page");
+        System.out.println("Step passed: browser title equals \"Home Page\"");
+
+
+        //3. Perform login
+        driver.findElement(By.xpath("/html/body/header/div/nav/ul[2]")).click();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Fixed.

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        driver.findElement(By.id("name")).sendKeys("Roman");
+        driver.findElement(By.id("password")).sendKeys("Jdi1234");
+        driver.findElement(By.xpath("//*[@id=\"login-button\"]")).click();
+        boolean actualLogIn = driver.findElement(By.xpath("/html/body/header/div/nav/ul[2]/li/div/div/button")).isEnabled();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Changed to class.

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        String currentTitle = driver.getTitle();
+        softAssert.assertEquals(currentTitle, "Home Page");
+        System.out.println("Step passed: browser title equals \"Home Page\"");
+
+
+        //3. Perform login
+        driver.findElement(By.xpath("/html/body/header/div/nav/ul[2]")).click();
+        driver.findElement(By.id("name")).sendKeys("Roman");
+        driver.findElement(By.id("password")).sendKeys("Jdi1234");
+        driver.findElement(By.xpath("//*[@id=\"login-button\"]")).click();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Fixed.

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        boolean actualLogIn = driver.findElement(By.xpath("/html/body/header/div/nav/ul[2]/li/div/div/button")).isEnabled();
+        softAssert.assertTrue(actualLogIn);
+        System.out.println("Step passed: user is logged");
+
+
+        //4. Assert Username is loggined
+        WebElement actualUserName = driver.findElement(By.xpath("//*[@id=\"user-name\"]"));
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Fixed.

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        String expectedUserName = "Roman Iovlev";
+        softAssert.assertEquals(actualUserName,expectedUserName);
+        System.out.println("Test passed: name is displayed and equals to expected result");
+
+
+        //5. Assert that there are 4 items on the header section are displayed and they have proper texts
+        List<WebElement> actualElements = driver.findElements(By.className("uui-navigation nav navbar-nav m-l8"));
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Fixed.

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        List<String> expectedElements = Arrays.asList("HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS");
+        softAssert.assertEquals(actualElements,expectedElements);
+        System.out.println("Step passed: menu buttons are displayed and have proper texts");
+
+
+        //6. Assert that there are 4 images on the Index Page and they are displayed
+        WebElement actualImage = driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div[2]"));
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Fixed.

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.*;
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.testng.asserts.SoftAssert;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class FunctionalTest1 {
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Fixed dublicates.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+    @BeforeTest
+    public void profileSetup() {
+        driver.manage().window().maximize();
+    }
+
+    @BeforeClass
+    public void appSetup() {
+        driver.get(url);
+    }
+
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+    @Test
+    public void FunctionalTest2() {
+        SoftAssert softAssert = new SoftAssert();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+
+
+        //2. Assert Browser title
+        String currentTitle = driver.getTitle();
+        softAssert.assertEquals(currentTitle, "Home Page");
+        System.out.println("Step passed: browser title equals \"Home Page\"");
+
+
+        //3. Perform login
+        driver.findElement(By.xpath("/html/body/header/div/nav/ul[2]")).click();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+        driver.findElement(By.id("name")).sendKeys("Roman");
+        driver.findElement(By.id("password")).sendKeys("Jdi1234");
+        driver.findElement(By.xpath("//*[@id=\"login-button\"]")).click();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+        boolean actualLogIn = driver.findElement(By.xpath("/html/body/header/div/nav/ul[2]/li/div/div/button")).isEnabled();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+        softAssert.assertTrue(actualLogIn);
+        System.out.println("Step passed: user is logged");
+
+
+        //4. Assert Username is loggined
+        WebElement actualUserName = driver.findElement(By.xpath("//*[@id=\"user-name\"]"));
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 17]
```diff
@@ -0,0 +1,117 @@
...
+        String expectedUserName = "Roman Iovlev";
+        softAssert.assertEquals(actualUserName,expectedUserName);
+        System.out.println("Step passed: name is displayed and equals to expected result");
+
+
+        //5.Open through the header menu Service -> Different Elements Page
+        driver.findElement(By.xpath("//*[@id=\"mCSB_1_container\"]/ul/li[3]/a")).click();
+        driver.findElement(By.xpath("//*[@id=\"mCSB_1_container\"]/ul/li[3]/ul/li[8]/a/span")).click();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 18]
```diff
@@ -0,0 +1,117 @@
...
+        String expectedPage = "https://jdi-testing.github.io/jdi-light/different-elements.html";
+        softAssert.assertEquals(actualPage,expectedPage);
+        System.out.println("Step passed: page is opened");
+
+
+        //6.Select checkboxes
+        driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div/div[2]/label[1]/input")).click();
+        driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div/div[2]/label[3]/input")).click();
+        boolean actualCheckBox1 = driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div/div[2]/label[1]/input")).isSelected();
+        boolean actualCheckBox2 = driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div/div[2]/label[3]/input")).isSelected();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 19]
```diff
@@ -0,0 +1,117 @@
...
+        softAssert.assertTrue(actualCheckBox1);
+        softAssert.assertTrue(actualCheckBox2);
+        System.out.println("Step passed: elements are checked");
+
+
+        //7.Select radio
+        driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div/div[3]/label[1]/input")).click();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 20]
```diff
@@ -0,0 +1,117 @@
...
+        boolean actualRadioButton = driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div/div[3]/label[1]/input")).isSelected();
+        softAssert.assertTrue(actualRadioButton);
+        System.out.println("Step passed: element is checked");
+
+
+        //8.Select in dropdown
+        driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div/div[4]/select")).click();
+        boolean actualDropDown = driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div/div[4]/select")).isSelected();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 21]
```diff
@@ -0,0 +1,117 @@
...
+        softAssert.assertTrue(actualDropDown);
+        System.out.println("Step passed: element is selected");
+
+
+        //9.Assert individual logging after selecting items
+        driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div/div[2]/label[1]/input")).click();
+        driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div/div[3]/label[1]/input")).click();
+        driver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div/div[4]")).click();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 22]
```diff
@@ -0,0 +1,117 @@
...
+
+        boolean actualLogging = driver.findElement(By.xpath("//*[@id=\"mCSB_2_container\"]/section[1]/div[2]")).isDisplayed();
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 23]
```diff
@@ -0,0 +1,117 @@
...
+        softAssert.assertTrue(actualLogging);
+        System.out.println("Step passed: log for all elements exist");
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed.

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 24]
```diff
@@ -0,0 +1,127 @@
...
+
+
+        //11. Assert that there are 5 items in the Left Section are displayed and they have proper text
+        List<WebElement> actualItems = driver.findElements(By.className("sidebar-menu"));
+        List<String> expectedItems = Arrays.asList(
+                "Home", "Contact form", "Service", "Metals & Colors", "Elements packs"
+        );
+        softAssert.assertEquals(actualItems,expectedItems);
+        System.out.println("Step passed: left section menu items are displayed and have proper text");
+
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Fixed.

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 25]
```diff
@@ -0,0 +1,117 @@
...
+        softAssert.assertEquals(actualUserName,expectedUserName);
+        System.out.println("Step passed: name is displayed and equals to expected result");
+
+
+        //5.Open through the header menu Service -> Different Elements Page
+        driver.findElement(By.xpath("//*[@id=\"mCSB_1_container\"]/ul/li[3]/a")).click();
+        driver.findElement(By.xpath("//*[@id=\"mCSB_1_container\"]/ul/li[3]/ul/li[8]/a/span")).click();
+        String actualPage = driver.getTitle();
+        String expectedPage = "https://jdi-testing.github.io/jdi-light/different-elements.html";
+        softAssert.assertEquals(actualPage,expectedPage);
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Rewrited this.

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 26]
```diff
@@ -0,0 +1,118 @@
...
+        List<String> expectedElements = Arrays.asList("HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS");
+        softAssert.assertEquals(actualElements,expectedElements);
+        System.out.println("Step passed: menu buttons are displayed and have proper texts");
+
+
+        //6. Assert that there are 4 images on the Index Page and they are displayed
+        WebElement actualImage = driver.findElement(By.className("benefit-icon"));
+        boolean expectedImage = actualImage.isDisplayed();
+        softAssert.assertEquals(actualImage,expectedImage);
+        System.out.println("Step passed: images are displayed");
```
Comment (DmitryKhodakovsky) [inline test/java/hw2/Ex1/FunctionalTest1.java:71]: How much images should be displayed?

File: test/java/hw2/Ex1/FunctionalTest1.java (hunk 24 above)
[the remark this answers is not in the document]
Comment (DmitryKhodakovsky) [reply test/java/hw2/Ex1/FunctionalTest1.java:109]: I do not get what was fixed. This test always has status **PASS**

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 27]
```diff
@@ -0,0 +1,118 @@
...
+        );
+        softAssert.assertEquals(actualItems,expectedItems);
+        System.out.println("Step passed: left section menu items are displayed and have proper text");
+
+    }
+
+    //12. Close browser
+    @AfterClass
+    public void tearDown() {
+        if(driver!=null)
```
Comment (DmitryKhodakovsky) [inline test/java/hw2/Ex1/FunctionalTest1.java:115]: Why do you need this `if`?

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 28]
```diff
@@ -0,0 +1,123 @@
...
+    private String url = "https://jdi-testing.github.io/jdi-light/index.html ";
+
+
+    @BeforeMethod
+    public void setUp() {
+        WebDriverManager.chromedriver().setup();
+        driver = new ChromeDriver();
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+        driver.manage().window().maximize();
+    }
```
Comment (DmitryKhodakovsky) [inline test/java/hw2/Ex2/FunctionalTest2.java:31]: Similar code in the `FunctionalTest1` class

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 29]
```diff
@@ -0,0 +1,123 @@
...
+        assertTrue(actualLogIn);
+        System.out.println("Step passed: user is logged");
+
+
+        //4. Assert Username is loggined
+
+        WebElement actualUserName = driver.findElement(By.id("user-name"));
+        assertTrue(actualUserName.isDisplayed());
+        System.out.println("Step passed: name is displayed and equals to expected result");
+
```
Comment (DmitryKhodakovsky) [inline test/java/hw2/Ex2/FunctionalTest2.java:62]: Assertion is missing. Please read carefully test steps

File: test/java/hw2/Ex2/FunctionalTest2.java [hunk 30]
```diff
@@ -0,0 +1,123 @@
...
+        assertTrue(waterLog.isDisplayed());
+
+        WebElement windLog = driver.findElement(By.xpath("//li[contains(text(),'Wind: condition changed to true')]"));
+        assertTrue(windLog.isDisplayed());
+
+        WebElement selenLog = driver.findElement(By.xpath("//li[contains(text(),'metal: value changed to  Selen')]"));
+        assertTrue(selenLog.isDisplayed());
+
+        WebElement dropDownLog = driver.findElement(By.xpath("//li[contains(text(),'Colors: value changed to Yellow')]"));
+        assertTrue(dropDownLog.isDisplayed());
```
Comment (DmitryKhodakovsky) [inline test/java/hw2/Ex2/FunctionalTest2.java:112]: What are you trying to check? Please read step.

File: test/java/hw2/Ex1/FunctionalTest1.java [hunk 31]
```diff
@@ -0,0 +1,118 @@
...
+    private WebDriver driver;
+    private String url = "https://jdi-testing.github.io/jdi-light/index.html";
+
+    @BeforeMethod
+    public void setUp() {
+        WebDriverManager.chromedriver().setup();
+        driver = new ChromeDriver();
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+        driver.manage().window().maximize();
+    }
```
[the remark this answers is not in the document]
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Fixed to one base test with open site, login and quit.

File: test/java/hw2/Ex1/FunctionalTest1.java (hunk 26 above)
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Four items. Fixed with wait.until.

File: test/java/hw2/Ex1/FunctionalTest1.java (hunk 27 above)
Comment (Innokentiy945) [reply test/java/hw2/Ex1/FunctionalTest1.java]: Rebased to main test witout if.

File: test/java/hw2/Ex2/FunctionalTest2.java (hunk 28 above)
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed to one base test with open site, login and quit.

File: test/java/hw2/Ex2/FunctionalTest2.java (hunk 30 above)
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: In this step i`m checking log massage after pressed button/checkbox/dropdown.

File: test/java/hw2/Ex2/FunctionalTest2.java (hunk 29 above)
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: There is a "assertTrue(actualUserName.isDisplayed());" 
Why it does not fit?

File: test/java/hw2/Ex2/FunctionalTest2.java (hunk 30 above)
Comment (DmitryKhodakovsky) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Why do you decide that created xpath based on the text good idea? 

Why do you do not check log text ?

File: src/test/java/hw2/Ex1/FunctionalTest1.java [hunk 32]
```diff
@@ -0,0 +1,96 @@
...
+
+
+        //8. Assert that there is the iframe with “Frame Button” exist
+        softAssert.assertTrue(driver.findElement(By.id("frame")).isDisplayed());
+        System.out.println("Step passed: the iframe exists");
+
+
+        //9. Switch to the iframe and check that there is “Frame Button” in the iframe
+        driver.switchTo().frame("frame");
+        softAssert.assertTrue(driver.findElement(By.id("button-frame")).isDisplayed());
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/Ex1/FunctionalTest1.java:77]: If I change this line to ` softAssert.assertTrue(!driver.findElement(By.id("button-frame")).isDisplayed());` test **passed**

Comment (Innokentiy945) [reply src/test/java/hw2/Ex1/FunctionalTest1.java:77]: Fixed for "softAssert.assertTrue(!driver.findElement(By.id("button-frame")).isDisplayed());" test passed. When i used past line in separate test there was an error.

File: test/java/hw2/Ex2/FunctionalTest2.java (hunk 30 above)
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Because it`s a simpliest way, to get page part.

I`m checking log with "assertTrue". Isn't that right?

Comment (DmitryKhodakovsky) [reply test/java/hw2/Ex2/FunctionalTest2.java]: If text of the log will changed your test will failed. You should have flexible locators

File: src/test/java/hw2/Ex1/FunctionalTest1.java (hunk 32 above)
Comment (DmitryKhodakovsky) [reply src/test/java/hw2/Ex1/FunctionalTest1.java]: What was fixed?

File: test/java/hw2/Ex2/FunctionalTest2.java (hunk 30 above)
Comment (Innokentiy945) [reply test/java/hw2/Ex2/FunctionalTest2.java]: Fixed to more flexible xpath.

File: src/test/java/hw2/Ex1/FunctionalTest1.java (hunk 32 above)
Comment (Innokentiy945) [reply src/test/java/hw2/Ex1/FunctionalTest1.java]: Rewrite this test with wait.until.

Also, i did this in previous step.

File: src/test/java/hw2/Ex2/FunctionalTest2.java [hunk 33]
```diff
@@ -0,0 +1,93 @@
...
+        assertTrue(waterLog.isDisplayed());
+
+        WebElement windLog = driver.findElement(By.xpath("//li[contains(text(),'Water') and contains(text(),'true')]"));
+        assertTrue(windLog.isDisplayed());
+
+        WebElement selenLog = driver.findElement(By.xpath("//li[contains(text(),'Selen') and contains(text(),'true')]"));
+        assertTrue(selenLog.isDisplayed());
+
+        WebElement dropDownLog = driver.findElement(By.xpath("//li[contains(text(),'Colors') and contains(text(),'Yellow')]"));
+        assertTrue(dropDownLog.isDisplayed());
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/Ex2/FunctionalTest2.java:89]: Why do you do not check text in the log element itself? How will you know that text in on the message correct?

File: src/test/java/hw2/Ex1/FunctionalTest1.java [hunk 34]
```diff
@@ -0,0 +1,99 @@
...
+        System.out.println("Step passed: driver has focus on the original page");
+
+
+        //11. Assert that there are 5 items in the Left Section are displayed and they have proper text
+        List<WebElement> actualItems = driver.findElements(By.className("sidebar-menu"));
+        List<String> expectedItems = Arrays.asList(
+                "Home", "Contact form", "Service", "Metals & Colors", "Elements packs"
+        );
+        softAssert.assertEquals(actualItems,expectedItems);
+        System.out.println("Step passed: left section menu items are displayed and have proper text");
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/Ex1/FunctionalTest1.java:94]: I still see that softAssertions does not work in this test.

File: src/test/java/hw2/Ex1/FunctionalTest1.java (hunk 32 above)
Comment (DmitryKhodakovsky) [reply src/test/java/hw2/Ex1/FunctionalTest1.java]: I could change any assertion in this test to incorrect and after that test will have status **PASSED**

**Comment by DmitryKhodakovsky** [comment]
Could be merged

**Pull request merged**